x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/Expression.java

```diff
@@ -50,6 +50,14 @@ public boolean resolved() {
             return failed == false;
         }
 
+        public TypeResolution and(TypeResolution other) {
+            if (this.failed == false) {
+                return other;
+            } else {
+                return this;
+            }
+        }
+
```

Comment:
`return failed ? this : other`

It's nice but I wonder if there's any value of having such a short method in.

Reply:
I also find the suggestion more readable. 
But also value in the method, otherwise chaining is more verbose, the caller having to check the failure state at every step.